File: Examples/ObjcDemo/RixSDKPodDemo/RixEngineObjcDemo/UI/TopOn/TopOnRewardVideoVC.m
#import "TopOnRewardVideoVC.h"
#import <AnyThinkSDK/AnyThinkSDK.h>
#import "AdConfig.h"

@interface TopOnRewardVideoVC () <ATAdLoadingDelegate, ATRewardedVideoDelegate>

@property (nonatomic, strong) UILabel *label;
@property (nonatomic, assign) BOOL isLoading;
@property (nonatomic, copy) NSString *TAG;

@end

@implementation TopOnRewardVideoVC

- (void)viewDidLoad {
    [super viewDidLoad];
    self.TAG = @"TopOn-rewardVideo:";
    self.view.backgroundColor = [UIColor whiteColor];
    self.navigationItem.title = NSLocalizedString(@"topOn_rewardVideo", @"");

    UIButton *bnLoad = [self createButtonWithTitle:NSLocalizedString(@"load_ad", @"") action:@selector(buttonLoad)];
    [self.view addSubview:bnLoad];

    UIButton *bnShow = [self createButtonWithTitle:NSLocalizedString(@"show_ad", @"") action:@selector(buttonShow)];
    [self.view addSubview:bnShow];

    self.label = [self createLabel];
    [self.view addSubview:self.label];

    [NSLayoutConstraint activateConstraints:@[
        [bnLoad.leadingAnchor constraintEqualToAnchor:self.view.leadingAnchor],
        [bnLoad.trailingAnchor constraintEqualToAnchor:self.view.trailingAnchor],
        [bnLoad.heightAnchor constraintEqualToConstant:50],
        [bnLoad.topAnchor constraintEqualToAnchor:self.view.safeAreaLayoutGuide.topAnchor],

        [bnShow.leadingAnchor constraintEqualToAnchor:self.view.leadingAnchor],
        [bnShow.trailingAnchor constraintEqualToAnchor:self.view.trailingAnchor],
        [bnShow.heightAnchor constraintEqualToConstant:50],
        [bnShow.topAnchor constraintEqualToAnchor:bnLoad.bottomAnchor constant:20],

        [self.label.leadingAnchor constraintEqualToAnchor:self.view.leadingAnchor],
        [self.label.trailingAnchor constraintEqualToAnchor:self.view.trailingAnchor],
        [self.label.heightAnchor constraintEqualToConstant:50],
        [self.label.topAnchor constraintEqualToAnchor:bnShow.bottomAnchor constant:20],
    ]];
}

- (void)buttonLoad {
    NSLog(@"bnLoad");
    if (self.isLoading) {
        return;
    }
    [self updateUI:YES message:NSLocalizedString(@"loading", @"")];
    [self loadAd];
}

- (void)updateUI:(BOOL)loading message:(NSString *)msg {
    self.isLoading = loading;
    self.label.text = msg;
}

- (void)buttonShow {
    if ([[ATAdManager sharedManager] rewardedVideoReadyForPlacementID:[AdConfig TopOn_Reward_Video_Ad_Id]]) {
        [[ATAdManager sharedManager] showRewardedVideoWithPlacementID:[AdConfig TopOn_Reward_Video_Ad_Id] inViewController:self delegate:self];
    } else {
        NSLog(@"Ad wasn't ready");
    }
}

- (void)loadAd {
    NSMutableDictionary *extra = [NSMutableDictionary dictionary];
    extra[kATAdLoadingExtraMediaExtraKey] = @"media_val_RewardedVC";
    [[ATAdManager sharedManager] loadADWithPlacementID:[AdConfig TopOn_Reward_Video_Ad_Id] extra:extra delegate:self];
}

#pragma mark - ATAdLoadingDelegate

- (void)didFinishLoadingADWithPlacementID:(NSString *)placementID {
    NSLog(@"%@ didFinishLoadingAD", self.TAG);
    [self updateUI:NO message:NSLocalizedString(@"load_success", @"")];
}

- (void)didFailToLoadADWithPlacementID:(NSString *)placementID error:(NSError *)error {
    NSString *msg = [NSString stringWithFormat:@"%ld: %@", (long)error.code, error.localizedDescription];
    NSLog(@"%@ didFailToLoadAD: %@", self.TAG, msg);
    [self updateUI:NO message:[NSString stringWithFormat:NSLocalizedString(@"load_failed", @""), msg]];
}

#pragma mark - ATRewardedVideoDelegate

- (void)rewardedVideoDidStartPlayingForPlacementID:(NSString *)placementID extra:(NSDictionary *)extra {
    NSLog(@"%@ rewardedVideoDidStartPlaying", self.TAG);
}

- (void)rewardedVideoDidEndPlayingForPlacementID:(NSString *)placementID extra:(NSDictionary *)extra {
    NSLog(@"%@ rewardedVideoDidEndPlaying", self.TAG);
}

- (void)rewardedVideoDidClickForPlacementID:(NSString *)placementID extra:(NSDictionary *)extra {
    NSLog(@"%@ rewardedVideoDidClick: click", self.TAG);
}

- (void)rewardedVideoDidCloseForPlacementID:(NSString *)placementID rewarded:(BOOL)rewarded extra:(NSDictionary *)extra {
    NSLog(@"%@ rewardedVideoDidClose: close", self.TAG);
}

- (void)rewardedVideoDidRewardSuccessForPlacemenID:(NSString *)placementID extra:(NSDictionary *)extra {
    NSLog(@"%@ rewardedVideoDidRewardSuccess: reward", self.TAG);
}

- (void)rewardedVideoDidFailToPlayForPlacementID:(NSString *)placementID error:(NSError *)error extra:(NSDictionary *)extra {
    NSLog(@"%@ rewardedVideoDidFailToPlay: play error: %@", self.TAG, error.localizedDescription);
}

@end

File: AlxAdsDemo/UI/TopOn/TopOnRewardVideoVC.swift
//
//  TopOnRewardVideoVC.swift
//  AlxAdsDemo
//
//  Created by liu weile on 2025/8/18.
//

import UIKit
import AnyThinkSDK
import AnyThinkRewardedVideo

class TopOnRewardVideoVC: BaseUIViewController {
    
    private let TAG = "TopOn-rewardVideo:"

    private var label:UILabel!

    private var isLoading:Bool=false

    private var rewardAd:ATRewardedVideo?

    override func viewDidLoad() {
        super.viewDidLoad()

        view.backgroundColor = .white
        navigationItem.title=NSLocalizedString("topOn_rewardVideo", comment: "")

        let bnLoad=createButton(title: NSLocalizedString("load_ad", comment: ""),  action: #selector(buttonLoad))
        view.addSubview(bnLoad)

        let bnShow=createButton(title: NSLocalizedString("show_ad", comment: ""),  action: #selector(buttonShow))
        view.addSubview(bnShow)

        label=createLabel()
        view.addSubview(label)

        NSLayoutConstraint.activate([
            bnLoad.leftAnchor.constraint(equalTo: view.leftAnchor),
            bnLoad.rightAnchor.constraint(equalTo: view.rightAnchor),
            bnLoad.widthAnchor.constraint(equalTo: view.widthAnchor),
            bnLoad.heightAnchor.constraint(equalToConstant: 50),
            bnLoad.topAnchor.constraint(equalTo: view.safeAreaLayoutGuide.topAnchor),

            bnShow.leftAnchor.constraint(equalTo: view.leftAnchor),
            bnShow.rightAnchor.constraint(equalTo: view.rightAnchor),
            bnShow.widthAnchor.constraint(equalTo: view.widthAnchor),
            bnShow.heightAnchor.constraint(equalToConstant: 50),
            bnShow.topAnchor.constraint(equalTo: bnLoad.bottomAnchor, constant: 20),

            label.leftAnchor.constraint(equalTo: view.leftAnchor),
            label.rightAnchor.constraint(equalTo: view.rightAnchor),
            label.widthAnchor.constraint(equalTo: view.widthAnchor),
            label.heightAnchor.constraint(equalToConstant: 50),
            label.topAnchor.constraint(equalTo: bnShow.bottomAnchor, constant: 20),
           
        ])
    }
    
    @objc func buttonLoad(){
        print("bnLoad")
        if isLoading {
            return
        }

        updateUI(true, NSLocalizedString("loading", comment: ""))

        loadAd()
    }
    
    func updateUI(_ loading:Bool,_ msg:String){
        self.isLoading=loading
        self.label.text=msg
    }

    
    @objc func buttonShow(){
        if ATAdManager.shared().rewardedVideoReady(forPlacementID: AdConfig.TopOn_Reward_Video_Ad_Id) {
            ATAdManager.shared().showRewardedVideo(withPlacementID: AdConfig.TopOn_Reward_Video_Ad_Id, in: self, delegate: self)
        }else{
            print("Ad wasn't ready")
        }
    }
    
    func loadAd() {
        var extra:[String:Any] = [:]
        extra[kATAdLoadingExtraMediaExtraKey]="media_val_RewardedVC"
        
        ATAdManager.shared().loadAD(withPlacementID: AdConfig.TopOn_Reward_Video_Ad_Id, extra: extra, delegate: self)
    }

}

extension TopOnRewardVideoVC:ATAdLoadingDelegate{
    
    func didFinishLoadingAD(withPlacementID placementID: String) {
        print("\(TAG) didFinishLoadingAD")
        updateUI(false, NSLocalizedString("load_success", comment: ""))
    }
    
    func didFailToLoadAD(withPlacementID placementID: String, error: (any Error)) {
        let msg="\(error.code): \(error.localizedDescription)"
        print("\(TAG) didFailToLoadAD: \(msg)")
        updateUI(false, String(format: NSLocalizedString("load_failed", comment: ""), msg))
    }
    
}

extension TopOnRewardVideoVC:ATRewardedVideoDelegate{
    
    func rewardedVideoDidStartPlaying(forPlacementID placementID: String, extra: [AnyHashable : Any]) {
        print("\(TAG) rewardedVideoDidStartPlaying")
    }
    
    func rewardedVideoDidEndPlaying(forPlacementID placementID: String, extra: [AnyHashable : Any]) {
        print("\(TAG) rewardedVideoDidEndPlaying")
    }
    
    func rewardedVideoDidClick(forPlacementID placementID: String, extra: [AnyHashable : Any]) {
        print("\(TAG) rewardedVideoDidClick: click")
    }
    
    func rewardedVideoDidClose(forPlacementID placementID: String, rewarded: Bool, extra: [AnyHashable : Any]) {
        print("\(TAG) rewardedVideoDidClose: close")
    }
    
    func rewardedVideoDidRewardSuccess(forPlacemenID placementID: String, extra: [AnyHashable : Any]) {
        print("\(TAG) rewardedVideoDidRewardSuccess: reward")
    }
    
    func rewardedVideoDidRewardSuccess(forPlacementID placementID: String, rewardAmount amount: Double) {
        print("\(TAG) rewardedVideoDidRewardSuccess: reward")
    }
    
    func rewardedVideoDidFailToPlay(forPlacementID placementID: String, error: any Error, extra: [AnyHashable : Any]) {
        print("\(TAG) rewardedVideoDidFailToPlay: play error: \(error.localizedDescription)")
    }
}
